From 5f79e89da8b8f1390876c9eb0fab4d67b150d0f5 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Fri, 17 Feb 2017 12:58:55 +0100
Subject: [PATCH 50/N] fixup! POSIX-ify the SHELL variable

Funnily enough, setting the SHELL variable to the empty string will be
mishandled without this fix when spawning a child process: the
build_env() method insists on using a temporary win_env for the
conversion that needs to be copied from the convvars (and hence wants
to copy the skip_if_empty bit, too).

It is unclear to this developer how this worked in 2.6.1-4 (it should
not have worked, but it clearly did, otherwise Git's test suite would
have failed in t3404-interactive-rebase's exec test which sets SHELL in
that very way), but it is an obviously correct bug fix in any case.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/environ.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/winsup/cygwin/environ.cc b/winsup/cygwin/environ.cc
index 00f9ec8..6ec050d 100644
--- a/winsup/cygwin/environ.cc
+++ b/winsup/cygwin/environ.cc
@@ -367,6 +367,7 @@ win_env::operator = (struct win_env& x)
   toposix = x.toposix;
   towin32 = x.towin32;
   immediate = false;
+  skip_if_empty = x.skip_if_empty;
   return *this;
 }
 
-- 
2.9.0

